### PR TITLE
#176 Feature Request: Add history to V2G Liberty entities

### PR DESCRIPTION
**Pull request by ArdJonker**
Done by adding the `car_state_of_charge`, `charger_real_charging_power`,  `car_remaining_range` sensors (and others) to the `v2g_liberty_package.yaml`. These were previously created by writing to it from AppDaemon and as such not known to HA and thus not in the history.

Remaining problems:
- All sensor entities have to be derived from a template (enforced by HA) and as they are populated by AppDaemon they don't have a relevant value_template, so a dummy of 0 is used. This results in a (brief) 0 value after restarts. It would be nice if this could be fixed later.
- O.a. the attribute `state_class: measurement` of the sensors cannot be used as it is a template.
- The attributes are not preserved as the set_state() calls overwrite attributes. 
A way to fix these problems might be to use mqtt.

Refactored several input_text and input_number entities to sensors so that (most) input_x entities are only used for actual user input. Renamed to `charger_charger_state` to `charger_state_int`, `charger_state` to `charger_state_text` and `charger_connected_car_state_of_charge` to `car_state_of_charge` for clarity and readability.

Also added a new sensor `current_scheduled_charging_power` that reflects the power that is currently requested by the schedule from the charger. This can help detecting a failing charger.

### CODE REVIEW

File: v2g-liberty/rootfs/root/appdaemon/apps/v2g_liberty/data_monitor.py [hunk 1]
```diff
@@ -94,7 +94,7 @@ async def initialize(self):
 
         await self.hass.listen_state(
             self.__handle_charger_state_change,
-            "sensor.charger_charger_state",
+            "sensor.charger_state_int",
```
[the remark this answers is not in the document]
Comment (ArdJonker) [reply v2g-liberty/rootfs/root/appdaemon/apps/v2g_liberty/data_monitor.py:97]: I agree, and i will, as usual, send an e-mail to all users where this is explained.

File: v2g-liberty/rootfs/root/appdaemon/apps/v2g_liberty/modbus_evse_client.py [hunk 2]
```diff
@@ -752,7 +757,7 @@ async def __get_car_soc(self, do_not_use_cache: bool = False) -> int:
         return state
 
     async def __get_charger_state(self) -> int:
-        self.__log("__get_charger_state")
+        self.__log("called")
```
[the remark this answers is not in the document]
Comment (ArdJonker) [reply v2g-liberty/rootfs/root/appdaemon/apps/v2g_liberty/modbus_evse_client.py:760]: I have the same feeling, on the other hand it is very convenient the __log method adds the scope so it does not have to be repeated (and is never forgotten).
Hoe could we have best of both worlds?

[the remark this answers is not in the document]
Comment (rhpijnacker) [reply v2g-liberty/rootfs/root/appdaemon/apps/v2g_liberty/modbus_evse_client.py:760]: What is the need for logging this? Should we create something like a "trace" helper function, that does exactly this?

[the remark this answers is not in the document]
Comment (ArdJonker) [reply v2g-liberty/rootfs/root/appdaemon/apps/v2g_liberty/modbus_evse_client.py:760]: Hmm, interesting! I'd like to discuss this!

File: v2g-liberty/rootfs/root/homeassistant/packages/v2g_liberty/v2g_liberty_package.yaml [hunk 3]
```diff
@@ -243,88 +243,185 @@ input_number:
...
+    # The platform: number would have been more suitable but HA does not support that
+    # The platform: template is not a prefect match but the only thing that could be hacked
+    # Other properties like state_class: measurement, native_value: int are not allowed
+    # for a template sensor and set via python.
+    sensors:
+      car_state_of_charge:
+        friendly_name: Car state of charge
+        device_class: battery
+        unit_of_measurement: "%"
+        value_template: "{{ 0 }}"  # Dummy
```
[the remark this answers is not in the document]
Comment (ArdJonker) [reply v2g-liberty/rootfs/root/homeassistant/packages/v2g_liberty/v2g_liberty_package.yaml:258]: Yep! Works! No more "jumps" in the history!

File: v2g-liberty/rootfs/root/homeassistant/packages/v2g_liberty/v2g_liberty_package.yaml [hunk 4]
```diff
@@ -243,88 +243,185 @@ input_number:
...
+      total_charging_cost_last_7_days:
+        friendly_name: Total charging costs last 7 days
+        device_class: monetary
+        unit_of_measurement: €
+        value_template: "{{ 0 }}"  # Dummy
+
+      total_charged_energy_last_7_days:
+        friendly_name: Total charged energy last 7 days
+        device_class: energy_storage
+        unit_of_measurement: "kWh"
```
[the remark this answers is not in the document]
Comment (ArdJonker) [reply v2g-liberty/rootfs/root/homeassistant/packages/v2g_liberty/v2g_liberty_package.yaml:289]: Agree, removed them.

File: v2g-liberty/rootfs/root/homeassistant/packages/v2g_liberty/v2g_liberty_package.yaml [hunk 5]
```diff
@@ -243,88 +243,185 @@ input_number:
...
+        friendly_name: Car state of charge
+        device_class: battery
+        unit_of_measurement: "%"
+        value_template: "{{ 0 }}"  # Dummy
+
+      charger_real_charging_power:
+        friendly_name: Charger real charging power
+        device_class: power
+        unit_of_measurement: W
+        value_template: "{{ 0 }}"  # Dummy
```
[the remark this answers is not in the document]
Comment (ArdJonker) [reply v2g-liberty/rootfs/root/homeassistant/packages/v2g_liberty/v2g_liberty_package.yaml:264]: Gives a yaml error in IDE..

**Review by rhpijnacker — APPROVED** [review]

**Pull request merged**